### PR TITLE
Subscription callbacks for acl and interface app modules

### PR DESCRIPTION
**Pull request by sachinholla**
* Implemented translateSubscribe() and processSubscribe() methods in AclApp and IntfApp. Mapping is based on the existing processGet() implementation. This may not be accurate since both the app modules have not been updated since long time. But it will be useful for testing the subscription feature.

* Current IntfApp supports only physcial ports in get/set operations. Same limitation is applicable for subscription as well.

* Added utilities for processing the path objects and help prepare the translateSubscribe() resposne.

* Added gotest helper functions to call translateSubscribe() and verify the responses. Added gotest cases to verify different openconfig-acl and openconfig-interfaces yang paths.

* Fixed incorrect paths & payloads used by the existing AclApp get/set test cases.

Part of gNMI Subscription enhancements described in https://github.com/sonic-net/SONiC/pull/1287

### CODE REVIEW

**Review by tomek-US — COMMENTED** [review]
Thx for all those changes.

It was a pleasure to review this code.

A few mostly cosmetic comments.

File: translib/path/path.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+			return false
+		}
+		for k, tv := range t.Key {
+			if pv, ok := p.Key[k]; ok {
+				if tv != "*" && tv != pv {
+					return false
+				}
+			} else {
+				return false
+			}
```
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/path/path.go:313]: moved all checks to a single `if` statement

File: translib/path/path_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+func testKeyWC(path string, exp bool) func(*testing.T) {
+	return func(t *testing.T) {
+		p := parsePath(t, path)
+		if StrHasWildcardKey(path) != exp {
+			t.Errorf("StrHasWildcardKey failed for \"%s\"", path)
+		}
+		if HasWildcardKey(p) != exp {
+			t.Errorf("HasWildcardKey failed for \"%s\"", path)
+			t.Errorf("p=(%v)", p)
```
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/path/path_test.go:67]: changed to log only `p`

File: translib/intf_app.go [hunk 3]
```diff
@@ -251,11 +253,137 @@ func (app *IntfApp) translateAction(dbs [db.MaxDB]*db.DB) error {
...
+	return nil
 }
 
 func (app *IntfApp) processSubscribe(req processSubRequest) (processSubResponse, error) {
-	return processSubResponse{}, tlerr.New("not implemented")
+	resp := processSubResponse{
+		path: req.path,
+	}
+	switch req.table.Name {
+	case "PORT", "PORT_TABLE", "COUNTERS_PORT_NAME_MAP":
```
[the remark this answers is not in the document]
Comment (tomek-US) [reply translib/intf_app.go:378]: Hmm. We already have examples of consts for table names. For example [here](https://github.com/sonic-net/sonic-mgmt-common/blob/72e2cecb34407815bb7e37b8d0fc1654f4d9e829/translib/acl_app.go#L41) and [here](https://github.com/sonic-net/sonic-mgmt-common/blob/72e2cecb34407815bb7e37b8d0fc1654f4d9e829/translib/intf_app.go#L51).

File: translib/intf_app.go [hunk 4]
```diff
@@ -251,11 +253,137 @@ func (app *IntfApp) translateAction(dbs [db.MaxDB]*db.DB) error {
...
+			return tlerr.InvalidArgs("not a valid ipv%d address: %s", ipV, addr)
+		}
+		addrDbPattern = addr + "/*"
+	} else if ipV == 4 {
+		addrDbPattern = "*.*.*.*/*"
+	} else {
+		addrDbPattern = "*:*/*" // ipv6 address will have at least 1 colon
+	}
+
+	nb.New().PathKey("ip", addr).Table(db.ConfigDB, "INTERFACE").Key(name, addrDbPattern)
```
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/intf_app.go]: changed

File: translib/intf_app.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/intf_app.go]: changed

File: translib/intf_app.go [hunk 5]
```diff
@@ -251,11 +253,137 @@ func (app *IntfApp) translateAction(dbs [db.MaxDB]*db.DB) error {
...
 
 func (app *IntfApp) processSubscribe(req processSubRequest) (processSubResponse, error) {
-	return processSubResponse{}, tlerr.New("not implemented")
+	resp := processSubResponse{
+		path: req.path,
+	}
+	switch req.table.Name {
+	case "PORT", "PORT_TABLE", "COUNTERS_PORT_NAME_MAP":
+		path.SetKeyAt(resp.path, 1, "name", req.key.Get(0))
+	case "INTERFACE":
```
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/intf_app.go]: changed

**Review by tomek-US — APPROVED** [review]
LGTM

**Pull request merged**